File: python/leetcode/binary-search.py
class Solution:
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l = 0
        r = len(nums)
        while l < r:
            m = l + (r - l) // 2
            if nums[m] > target:
                r = m
            elif nums[m] == target:
                return m
            else:
                l = m + 1
        return -1

File: java/leetcode/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length;
        while (l < r) {
            int m = Math.round(l + (r - l) / 2);
            if (nums[m] > target) {
                r = m;
            } else if (nums[m] == target) {
                return m;
            } else {
                l = m + 1;
            }
        }
        return -1;
    }
}
